### PR TITLE
Add GitHub actions to build UI and API images

### PR DESCRIPTION
**Pull request by Tomcli**
UI and API images will be pushed on every new commit to `main` branch after this PR.

### CODE REVIEW

File: .github/workflows/mlx-ci.yml [hunk 1]
```diff
@@ -0,0 +1,81 @@
+name: KFP Tekton Unit Tests
```
Comment (ckadner) [inline .github/workflows/mlx-ci.yml:1]:
```YAML
name: "Build MLX API and UI images"
```

**Review by ckadner — CHANGES_REQUESTED** [review]
Thanks @Tomcli -- looks good other than the copy-pasted `name` ;-)

**Comment by Tomcli** [comment]
FYI Only commits to the `main` branch will trigger the jobs with `github.event_name == 'push'`, so jobs in the PR will not push to the dockerhub repo.

File: .github/workflows/mlx-ci.yml (hunk 1 above)
Comment (Tomcli) [reply .github/workflows/mlx-ci.yml]: thanks, fixed it.

**Review by ckadner — APPROVED** [review]
/lgtm

**Comment by Tomcli** [comment]
Example push job in my fork https://github.com/Tomcli/mlx/runs/3236092498?check_suite_focus=true

**Comment by Tomcli** [comment]
Looks like the github API doesn't like my commits. Let me force push the history.

**Comment by Tomcli** [comment]
Ok this is the glitch from the tide code, need to close and reopen PR.

**Comment by Tomcli** [comment]
There's some caching glitch in tide. I will manually merge only for this time.

**Pull request merged**